File: JAVA/WavePrintMatrixProgram.java
//Wave Print Matrix Program - Mr. Eshank Tyagi
import java.util.*;
class WavePrintMatrixProgram
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);

        System.out.print("\nEnter the number of rows : ");
        int r = sc.nextInt();

        System.out.print("\nEnter the number of columns : ");
        int c = sc.nextInt();

        int[][] arr = new int[r][c];

        System.out.println("\nEnter the elements of the matrix :-");

        for(int i = 0; i < r; i++)
        {
            for(int j = 0; j < c; j++)
            {
                arr[i][j] = sc.nextInt();
            }
        }

        System.out.println("The matrix is :-");

        for(int i = 0; i < r; i++)
        {
            for(int j = 0; j < c; j++)
                System.out.print(arr[i][j] + " ");

            System.out.println();
        }

        System.out.println("The wave print of the matrix is :-");

        for(int i = 0; i < c; i++)
        {
            if(i % 2 == 0)
            {
                for(int j = 0; j < r; j++)
                    System.out.print(arr[j][i] + " ");
            }
            else
            {
                for(int j = r - 1; j >= 0; j--)
                    System.out.print(arr[j][i] + " ");
            }
        }
        sc.close();
    }
}

File: C++/WavePrintMatrixProgram.cpp
//Wave Print Matrix Program - Mr. Eshank Tyagi
#include <iostream>
#include <vector>

int main() 
{
    std::vector<std::vector<int>> arr;
    int r, c;

    std::cout << "\nEnter the number of rows : ";
    std::cin >> r;

    std::cout << "\nEnter the number of columns : ";
    std::cin >> c;

    arr.resize(r, std::vector<int>(c));

    std::cout << "\nEnter the elements of the matrix :-\n";

    for(int i = 0; i < r; i++) 
    {
        for(int j = 0; j < c; j++) 
        {
            std::cin >> arr[i][j];
        }
    }

    std::cout << "The matrix is :-\n";

    for(int i = 0; i < r; i++) 
    {
        for(int j = 0; j < c; j++) 
        {
            std::cout << arr[i][j] << " ";
        }
        std::cout << std::endl;
    }

    std::cout << "The wave print of the matrix is :-\n";

    for(int i = 0; i < c; i++) 
    {
        if(i % 2 == 0) {
            for(int j = 0; j < r; j++) 
            {
                std::cout << arr[j][i] << " ";
            }
        }
        else 
        {
            for(int j = r - 1; j >= 0; j--) 
            {
                std::cout << arr[j][i] << " ";
            }
        }
    }

    return 0;
}
